### PR TITLE
py-cppy: add 1.2.1

### PR DESCRIPTION
**Pull request by manuelakuhn**
https://github.com/nucleic/cppy/tree/1.2.1

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-cppy/package.py [hunk 1]
```diff
@@ -14,7 +14,11 @@ class PyCppy(PythonPackage):
 
     maintainers = ["iarspider"]
 
+    version("1.2.1", sha256="83b43bf17b1085ac15c5debdb42154f138b928234b21447358981f69d0d6fe1b")
     version("1.1.0", sha256="4eda6f1952054a270f32dc11df7c5e24b259a09fddf7bfaa5f33df9fb4a29642")
 
+    depends_on("python@3.7:", when="@1.2:", type=("build", "run"))
     depends_on("python@3:", type=("build", "run"))
+    depends_on("py-setuptools@42:", when="@1.2:", type="build")
```
Comment (iarspider) [inline var/spack/repos/builtin/packages/py-cppy/package.py]:
```suggestion
    depends_on("py-setuptools@61.2:", when="@1.2:", type="build")
```

File: var/spack/repos/builtin/packages/py-cppy/package.py [hunk 2]
```diff
@@ -14,7 +14,11 @@ class PyCppy(PythonPackage):
...
     depends_on("py-setuptools", type="build")
+    depends_on("py-setuptools-scm+toml@3.4.3:", when="@1.2:", type="build")
```
Comment (iarspider) [inline var/spack/repos/builtin/packages/py-cppy/package.py:24]:
```suggestion
    depends_on("py-setuptools-scm+toml@3.4.3: +toml", when="@1.2:", type="build")
```

**Review by iarspider — CHANGES_REQUESTED** [review]

File: var/spack/repos/builtin/packages/py-cppy/package.py (hunk 1 above)
Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-cppy/package.py]: Thanks.

File: var/spack/repos/builtin/packages/py-cppy/package.py (hunk 2 above)
Comment (manuelakuhn) [reply var/spack/repos/builtin/packages/py-cppy/package.py:24]: `+toml` would be added twice then. Did you miss the one before the `@`?

Comment (iarspider) [reply var/spack/repos/builtin/packages/py-cppy/package.py:24]: @manuelakuhn yup. Sorry!

**Review by iarspider — APPROVED** [review]

**Comment by iarspider** [comment]
LGTM.

**Review by tldahlgren — COMMENTED** [review]
Confirmed sha256.

**Review by tldahlgren — APPROVED** [review]

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**